horovod/common/global_state.h

```diff
@@ -146,6 +146,12 @@ struct HorovodGlobalState {
   // A string indicating what framework we are using to perform CPU operations.
   std::string cpu_operation;
 
+  // Number of ranks that did Join()
+  int joined_size = 0;
+  bool joined = false;
```

Comment:
Can you add comment clarifying what does `joined` do and what does it mean if it's true or false?